homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -379,16 +353,16 @@ def _extract_value_from_attribute(state, attribute):
     @callback
     def _handle_coordinator_update(self):
         """Fetch state from the device."""
-        self._available = True
-        self._state = self.coordinator.data.is_on
-        self._state_attrs.update(
+        self._attr_available = True
+        self._attr_is_on = self.coordinator.data.is_on
+        self._attr_extra_state_attributes.update(
             {
                 key: self._extract_value_from_attribute(self.coordinator.data, value)
                 for key, value in self._available_attributes.items()
             }
         )
-        self._mode = self._state_attrs.get(ATTR_MODE)
-        self._fan_level = self._state_attrs.get(ATTR_FAN_LEVEL)
+        self._mode = self._attr_extra_state_attributes.get(ATTR_MODE)
```

Comment:
Maybe `self._mode = self.coordinator.data.mode` and remove `mode` attribute?

Reply:
I tried to do a simple mass-conversion without touching the behavior too much, so I'd have to look into this a bit as I don't know the code that well.